tasks/eth/006-MCP-L1/SignFromJsonWithAssertions.s.sol

```diff
@@ -0,0 +1,296 @@
+
+        require(proxies.OptimismPortal.code.length != 0, "5700");
+
+        OptimismPortal portal = OptimismPortal(payable(proxies.OptimismPortal));
+
+        address guardian = cfg.superchainConfigGuardian();
+
+        require(address(portal.L2_ORACLE()) == proxies.L2OutputOracle, "5800");
+        require(address(portal.l2Oracle()) == proxies.L2OutputOracle, "5900");
+        require(address(portal.SYSTEM_CONFIG()) == proxies.SystemConfig, "6000");
+        require(address(portal.systemConfig()) == proxies.SystemConfig, "6100");
+        require(portal.GUARDIAN() == guardian, "6200");
+        require(portal.guardian() == guardian, "6300");
+        require(address(portal.superchainConfig()) == address(proxies.SuperchainConfig), "6400");
+        require(portal.paused() == SuperchainConfig(proxies.SuperchainConfig).paused(), "6500");
+        require(portal.l2Sender() == Constants.DEFAULT_L2_SENDER, "6600");
+    }
+
+    /// @notice Asserts that the ProtocolVersions is setup correctly
+    function checkProtocolVersions() internal view {
+        console.log("Running chain assertions on the ProtocolVersions");
+
+        require(proxies.ProtocolVersions.code.length != 0, "6700");
+
+        ProtocolVersions versions = ProtocolVersions(proxies.ProtocolVersions);
+
+        require(versions.owner() == cfg.finalSystemOwner(), "6800");
+
+        string memory json;
+
+        try vm.readFile(string.concat(vm.projectRoot(), "/tasks/eth/005-protocol-versions-ecotone/input.json"))
+        returns (string memory data) {
+            json = data;
+        } catch {
+            revert("Failed to read tasks/eth/005-protocol-versions-ecotone/input.json");
+        }
+
+        require(
+            ProtocolVersion.unwrap(versions.required())
+                == uint256(stdJson.readBytes32(json, string(abi.encodePacked("$.transactions[0].data")))),
+            "6900"
+        );
+        require(
+            ProtocolVersion.unwrap(versions.recommended())
+                == uint256(stdJson.readBytes32(json, string(abi.encodePacked("$.transactions[1].data")))),
+            "7000"
+        );
+    }
+
+    /// @notice Asserts that the SuperchainConfig is setup correctly
+    function checkSuperchainConfig() internal view {
+        console.log("Running chain assertions on the SuperchainConfig");
+
+        require(proxies.SuperchainConfig.code.length != 0, "7100");
+
+        SuperchainConfig superchainConfig = SuperchainConfig(proxies.SuperchainConfig);
+
+        require(superchainConfig.guardian() == cfg.superchainConfigGuardian(), "7200");
+        require(superchainConfig.paused() == false, "7300");
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck() internal view override {
+        console.log("Running post-deploy assertions");
+        ResourceMetering.ResourceConfig memory actualResourceConfig =
+            SystemConfig(proxies.SystemConfig).resourceConfig();
```

Comment:
Why is the resource config checked here instead of inside of `checkSystemConfig`?